packages/core/src/model/TransactionStatus.ts

```diff
@@ -0,0 +1,11 @@
+import { ChainId } from '../constants'
+import { TransactionResponse, TransactionReceipt } from '@ethersproject/abstract-provider'
+
+export interface TransactionStatus {
+  status: 'None' | 'Mining' | 'Success' | 'Fail' | 'Exception'
+  errored: boolean
+  transaction?: TransactionResponse
+  receipt?: TransactionReceipt
+  chainId?: ChainId
+  errorMessage?: string
+}
```

Comment:
With `errored` property you can now do discriminated union like:
```ts
{errored: true; errorMessage: string } | {errored: false; errorMessage: never}
```

Reply:
Yea, just my own opinion... They looks same to me

Reply:
All I'm saying is without the boolean, the same could still be achieved...

Reply:
And, with just one way, there's surely a unified approach than having multiple variables doing same thing, it gives dev users a kind of fear both don't work the same way or they meant something different from development perspective...

Might just be a bad opinion rather

Reply:
After some discussion, I think it is fine approach (the whole type safety approach is confusing for many people and leads to hacking code every now an then). Let describe it correctly in changeset and PR name

Reply:
My argument is not against the boolean added, more against duplication of errorMessage, which in turn means everywhere TransactionStatus is initialized, you have to set errored|failed to false instead of just not bothering and not initializing errorMessage